sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceUtils.scala

```diff
@@ -58,6 +58,22 @@ object DataSourceUtils {
     Serialization.read[Seq[String]](str)
   }
 
+  /**
+   * Verify if the field name is supported in datasource. This verification should be done
+   * in a drover side.
+   */
+  def checkFieldNames(format: FileFormat, schema: StructType): Unit = {
```

Comment:
shall we merge this into `verifySchema`? The `schema: StructType` contains field names as well.